core/controllers/reader.py

```diff
@@ -597,6 +599,13 @@ def post(self, exploration_id):
             learner_progress_services.mark_collection_as_incomplete(
                 user_id, collection_id)
 
+        if user_id and story_id:
+            story = story_fetchers.get_story_by_id(story_id)
```

Comment:
Handle the case where story or topic was returned None (deleted) - don't do anything for this case.

Reply:
Done. PTAL!